bots/src/liquidator.rs

```diff
@@ -0,0 +1,418 @@
+    pub signer_account: AccountId,
+    /// Asset to liquidate
+    #[arg(short, long, env = "ASSET_ACCOUNT_ID")]
+    pub asset: AccountId,
+    /// Network to run liquidations on
+    #[arg(short, long, env = "NETWORK", default_value_t = Network::Testnet)]
+    pub network: Network,
+    /// Timeout for transactions
+    #[arg(short, long, env = "TIMEOUT", default_value_t = 60)]
+    pub timeout: u64,
+    /// Interval between liquidation attempts
+    #[arg(short, long, env = "INTERVAL", default_value_t = 600)]
+    pub interval: u64,
+    /// Concurency for liquidations
+    #[arg(short, long, env = "CONCURRENCY", default_value_t = 10)]
+    pub concurrency: usize,
+}
+
+pub struct Liquidator<S: Swap> {
+    client: JsonRpcClient,
+    signer: InMemorySigner,
+    asset: AccountId,
+    pub market: AccountId,
+    timeout: u64,
+    swap: S,
+}
+
+impl<S: Swap> Liquidator<S> {
+    #[must_use]
+    pub fn new(
+        client: JsonRpcClient,
+        signer: InMemorySigner,
+        asset: AccountId,
+        market: AccountId,
+        swap: S,
+        timeout: u64,
+    ) -> Self {
+        Self {
+            client,
+            signer,
+            asset,
+            market,
+            timeout,
+            swap,
+        }
+    }
+
+    #[instrument(skip(self), level = "debug")]
+    async fn get_borrow_status(
+        &self,
+        borrow: AccountId,
+        oracle_response: &OracleResponse,
+    ) -> anyhow::Result<Option<BorrowStatus>> {
+        view(
+            &self.client,
+            self.market.clone(),
+            "get_borrow_status",
+            &json!({
+                "account_id": borrow,
+                "oracle_response": oracle_response,
+            }),
+        )
+        .await
+    }
+
+    fn create_transfer_tx(
+        &self,
+        borrow: &AccountId,
+        liquidation_amount: U128,
+        nonce: u64,
+        block_hash: CryptoHash,
+    ) -> Transaction {
+        #[allow(clippy::unwrap_used, reason = "We know the serialization will succeed")]
+        let msg = serde_json::to_string(&DepositMsg::Liquidate(LiquidateMsg {
+            account_id: borrow.clone(),
+        }))
+        .unwrap();
+
+        Transaction::V0(TransactionV0 {
+            nonce,
+            receiver_id: self.asset.clone(),
+            block_hash,
+            signer_id: self.signer.account_id.clone(),
+            public_key: self.signer.public_key().clone(),
+            actions: vec![Action::FunctionCall(Box::new(FunctionCallAction {
+                method_name: "ft_transfer_call".to_string(),
+                args: serialize_and_encode(json!({
+                    "receiver_id": self.market,
+                    "amount": liquidation_amount,
+                    "msg": msg,
+                })),
+                gas: DEFAULT_GAS,
+                deposit: ONE_YOCTO_NEAR,
+            }))],
+        })
+    }
+
+    #[instrument(skip(self), level = "debug")]
+    pub async fn try_liquidate(
```

Comment:
Rename this to `liquidate` as `try_` isn't idiomatic Rust. It used to be popular in very early Rust, but these days its never used.